File: admin/src/types/room.ts
export interface Room {
  id: string
  name: string
  building: string | null
  capacity: number | null
  camera_endpoint: string | null
  /**
   * mediamtx path the room publishes to (e.g. "eb226"). Derivable from
   * `camera_endpoint` (`rtsp://host:port/{stream_key}`) but the backend
   * also stores it as a first-class field so the live-feed code doesn't
   * have to re-parse the URL on every request.
   */
  stream_key: string | null
  is_active: boolean
}

export interface RoomCreate {
  name: string
  building?: string
  capacity?: number
  camera_endpoint?: string
}

export interface RoomUpdate {
  name?: string
  building?: string
  capacity?: number
  camera_endpoint?: string
  is_active?: boolean
}

File: backend/app/schemas/room.py
"""
Room Schemas

Request and response models for room operations.
"""

from pydantic import BaseModel, Field


class RoomCreate(BaseModel):
    """Room creation request"""

    name: str = Field(..., min_length=1, max_length=100)
    building: str = Field(..., min_length=1, max_length=100)
    capacity: int | None = None
    camera_endpoint: str | None = Field(None, max_length=255)


class RoomUpdate(BaseModel):
    """Room update request (all fields optional)"""

    name: str | None = Field(None, min_length=1, max_length=100)
    building: str | None = Field(None, min_length=1, max_length=100)
    capacity: int | None = None
    camera_endpoint: str | None = Field(None, max_length=255)
    is_active: bool | None = None
